### PR TITLE
Restore Halogen recipes

### LINKED ISSUE #203: Fix DriverWebSocketsHalogenHooks in TPS
The error in the console is:
> Uncaught DOMException: The operation is insecure.
    create Web.Socket.WebSocket/foreign.js:8

[try](https://try.ps.ai/?github=JordanMartinez/purescript-cookbook/master/recipes/DriverWebSocketsHalogenHooks/src/Main.purs)

Not obvious how to fix this one.

### PR DESCRIPTION
**Pull request by pete-murphy**
Fixes #203 

Restoring some broken Halogen recipes. The `DriverWebSocketsHalogenHooks` is adapted from the work in this PR https://github.com/purescript-halogen/purescript-halogen/pull/804, consolidated to a single component to be TryPureScript-friendly, and using the `-hooks` version of the component from the original recipe. Aside from that, the other recipes were fixed by just following the guidance of the changelog https://github.com/purescript-halogen/purescript-halogen/blob/master/docs/changelog/v6.md.

Also made an unrelated change to fix the naming of a function in one of the `DogImages` recipes (`randomCatGif` → `randomDogImage`) [42d3448](https://github.com/JordanMartinez/purescript-cookbook/pull/306/commits/42d34482c288a88e0a130eb0cf55f9c05e2f4d42)

### CODE REVIEW

File: recipes/AceEditorHalogenHooks/web/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"use strict";
```
Comment (JordanMartinez) [inline recipes/AceEditorHalogenHooks/web/index.js:1]: Can we get rid of 'use strict' here?

Comment (pete-murphy) [reply recipes/AceEditorHalogenHooks/web/index.js]: Yep, it's not needed 👍

Comment (pete-murphy) [reply recipes/AceEditorHalogenHooks/web/index.js]: [64c3836](https://github.com/JordanMartinez/purescript-cookbook/pull/306/commits/64c383647c2b200326b27e4a23d58c37812d04b8)

**Review by JordanMartinez — APPROVED** [review]

**Comment by JordanMartinez** [comment]
Thanks!

**Pull request merged**